assets/externalized/scripts/enums.lua

```diff
@@ -1,17 +1,195 @@
 Items = {
     SUNGOGGLE = 212,
-    CANTEEN   = 214
+    CANTEEN   = 214,
 }
 
 Visibility = {
 	BURIED               =  -3,
 	HIDDEN_IN_OBJECT     =  -2,
 	INVISIBLE            =  -1,
-	VISIBLE              =   1
+	VISIBLE              =   1,
 }
 
 SectorFlags = {
     SF_ALREADY_VISITED = 32,
     SF_USE_ALTERNATE_MAP = 64,
-    SF_HAS_ENTERED_TACTICAL = 512
+    SF_HAS_ENTERED_TACTICAL = 512,
+}
+
+CivGroups = {
+    NON_CIV_GROUP = 0,
+    REBEL_CIV_GROUP = 1,
+    KINGPIN_CIV_GROUP = 2,
+    SANMONA_ARMS_GROUP = 3,
+    ALMA_MILITARY_CIV_GROUP = 7,
+    DOCTORS_CIV_GROUP = 8,
+    HICKS_CIV_GROUP = 10,
+    WARDEN_CIV_GROUP = 11,
+}
+
+Teams = {
+    OUR_TEAM = 0,
+    ENEMY_TEAM = 1,
+    CREATURE_TEAM = 2,
+    MILITIA_TEAM = 3,
+    CIV_TEAM = 4,
+}
+
+SoldierClasses = {
```

Comment:
I wonder if there is a better way to define enums that are already defined in C++. This looks quite error prone...

Reply:
I would think about code generation. Do you know of any good ways and tools (that understands C code) for that?

Reply:
Unfortunately not. But I found there is a `new_enum` function in sol, which we could use from C++ to create the enum: https://sol2.readthedocs.io/en/latest/api/table.html#new-enum

Reply:
If I am not mistaken, with the `new_enum` function we still need to provide each and every enum name and value. This is just as error-prone and copy-and-pasting.

I am thinking if we can get away with a simple'ish regex scripts, that specifically handles our usage of `enum {` and `#define SOME_PREFIX_` - oh yes, some of the "enums" are a series of `define` sharing the same prefix.

Reply:
Just found this: https://github.com/Neargye/magic_enum

But requires C++17, and we also need to change those `#define` into `enum`.

Reply:
That looks quite nice, but probably requires us to migrate all enums to enum classes, so I think it is out of scope for this PR. What is keeping us from using C++17? Only MinGW on travis? Because I wanted to suggest to retire Travis as we already have Github Actions which give us excactly the same functionality. Only nightly builds would have to be migrated, which is trivial imho.

Reply:
One problem is `rapidjson`. The library gives warnings on C++14 and C++ 17 (hard errors on VC++ 2019). There is a [fix](https://github.com/Tencent/rapidjson/issues/1131) on master already but there has been no new releases.

We might need to pin to a newer commit on master, or switch to another library.